content/concepts/complex-payment-types/escrow.md

```diff
@@ -33,12 +33,32 @@ Escrow is designed as a feature to enable the XRP Ledger to be used in the [Inte
 - Escrow only works with XRP, not issued currencies.
 - Escrow requires sending at least two transactions: one to create the escrow, and one to finish or cancel it. Thus, it may not be financially sensible to escrow payments for very small amounts, because the participants must destroy the [transaction cost](transaction-cost.html) of the two transactions.
     - When using Crypto-Conditions, the [cost of the transaction to finish the escrow](#escrowfinish-transaction-cost) is higher than usual.
-- All escrows must have a "finish-after" time, an expiration time, or both. Neither time can be in the past when the transaction to create the escrow executes.
+- All escrows must be created with a "finish-after" time or a [crypto-condition][]. If the escrow does not have a finish-after time, it must have an expiration time.
```

Comment:
This is true, but I'd phrase both conditions the same way. I.e `All escrows must be created with a finish-after time or an expiration time.`. In both cases we have a logical or, but in one case you phrased it with an `if` statement.

Reply:
It's complicated, because the following combos are all valid:
- FinishAfter plus any combination of CancelAfter and Condition
- Both Condition & CancelAfter

The following combinations are not valid:
- CancelAfter only (post-fix1571)
- Condition only

I couldn't find a more concise way of expressing the full set of cases than the "if" statement here.

Reply:
Here are four equavalent ways of saying this (if we consider english to be boolean logic, which I know it isn't):

1) What's there now:
```
All escrows must be created with a "finish-after" time or a crypto-condition. If the escrow does not have a "finish-after" time, it must have an expiration time.
```

2) Rephrase so both conditions use "or":
```
All escrows must be created with a "finish-after" time or a crypto-condition. All escrows must also be created with a "finish-after" time or an expiration time.
```

3) Rephrase so we group the "finish-after" time as the condition to an "if" statement.
```
If the escrow does not have a "finish-after" time, it must have *both* a cancel after time and a cryptocondition.
```

4) Rephrase so we group "finish-after" time as an "or" statement
```
All escrows must be created with a "finish-after" time or *both* a cancel after time and a cryptocondition.
```

None of these are prefect; I would probably vote for (4). But what's there is now is fine with me if you like that best.

Reply:
I think this is a case where treating English like boolean logic doesn't work very well. Making it two "or" statements both including the "finish-after" time in the two successive statements raises questions, for me, like, "Is that a typo? Are there two separate parts that have different requirements?"

I think the "or" followed by "if" works best in English. You're right that it's better to be clear that it's an inclusive or. (In general, I treat "or" in English as inclusive unless the text says "either ... or ..." but there's no harm in adding "or both" here.)

Reply:
Cool. FYI, still :+1: after latest changes.